dotnet-source/ks.fiks.io.arkivsystem.sample/ArkivSimulator.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Schema;
+using KS.Fiks.IO.Arkiv.Client.ForenkletArkivering;
+using KS.Fiks.IO.Arkiv.Client.Models;
+using KS.Fiks.IO.Arkiv.Client.Models.Arkivering.Arkivmelding;
+using KS.Fiks.IO.Arkiv.Client.Models.Arkivering.Arkivmeldingkvittering;
+using KS.Fiks.IO.Arkiv.Client.Models.Innsyn.Sok;
+using KS.Fiks.IO.Arkiv.Client.Models.Metadatakatalog;
+using ks.fiks.io.arkivintegrasjon.common.AppSettings;
+using ks.fiks.io.arkivintegrasjon.common.FiksIOClient;
+using ks.fiks.io.arkivsystem.sample.Generators;
+using ks.fiks.io.arkivsystem.sample.Handlers;
+using ks.fiks.io.arkivsystem.sample.Helpers;
+using ks.fiks.io.arkivsystem.sample.Models;
+using ks.fiks.io.arkivsystem.sample.Storage;
+using ks.fiks.io.arkivsystem.sample.Validering;
+using KS.Fiks.IO.Client;
+using KS.Fiks.IO.Client.Models;
+using KS.Fiks.IO.Client.Models.Feilmelding;
+using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Serilog;
+using EksternNoekkel = KS.Fiks.IO.Arkiv.Client.Models.Arkivering.Arkivmeldingkvittering.EksternNoekkel;
+
+namespace ks.fiks.io.arkivsystem.sample
+{
+    public class ArkivSimulator : BackgroundService
+    {
+        public const string TestSessionIdHeader = "testSessionId";
+        private FiksIOClient client;
+        private readonly AppSettings appSettings;
+        private static readonly ILogger Log = Serilog.Log.ForContext(MethodBase.GetCurrentMethod()?.DeclaringType);
+        public static SizedDictionary<string, Arkivmelding> _arkivmeldingCache;
+
+        public ArkivSimulator(AppSettings appSettings)
+        {
+            this.appSettings = appSettings;
+            Log.Information("Setter opp FIKS integrasjon for arkivsystem");
+            client = FiksIOClientBuilder.CreateFiksIoClient(appSettings);
+            _arkivmeldingCache = new SizedDictionary<string, Arkivmelding>(100);
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            Log.Information("ArkivSimulator is starting");
+            SubscribeToFiksIOClient();
+            await Task.CompletedTask;
```

Comment:
Kan fjerne async fra signatur med:
```suggestion
            return Task.CompletedTask;
```